nibabel/fileslice.py

```diff
@@ -54,6 +54,10 @@ def is_fancy(sliceobj):
         # slice or Ellipsis or None OK for  basic
         if isinstance(slicer, slice) or slicer in (None, Ellipsis):
             continue
+        # Allow single items in a dimension to be sliced
```

Comment:
So here you're allowing a single element array to be considered as an integer - could you say a bit more about why you need this?

I think fancy indexing always does a copy - I don't know whether the user will be surprised if this particular fancy indexing does not:

```
In [14]: a = np.arange(3)

In [15]: b = a[[0]]

In [16]: b
Out[16]: array([0])

In [17]: b[0] = 99

In [18]: a
Out[18]: array([0, 1, 2])

In [20]: c = a[0:1]

In [21]: c
Out[21]: array([0])

In [22]: c[0] = 99

In [23]: a
Out[23]: array([99,  1,  2])
```

Reply:
I'm allowing a single element list/vector to be considered as a slice, not as an integer. I'm just testing whether the thing in it can be treated as an integer. The reason is that if I want to take a single slice of an image while preserving the dimensions, the expected syntax would be for `img.slicer[[i]]` to be equivalent to `img.slicer[i:i+1]`.

Again, I have only been considering the expectations for result shape. That's a valid point about copying. I'm not sure about the best way to resolve this.